### ISSUE TITLE
Rule Change: no-unused-expressions should exclude statements joined by logical operators

### ISSUE THREAD

**Issue by stwlam** [eslint/eslint#17888]
### What rule do you want to change?

no-unused-expressions

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
function doStuff() {
   // ...
   const success = true; // or false
   return success;
}

function doOtherStuffInstead() {
   // stuff
}

doStuff() || doOtherStuffInstead();
```

### What does the rule currently do for this code?

The standard ("Expected an assignment or function call and instead saw an expression") warning is thrown.

### What will the rule do after it's changed?

Silence

### Participation

- [X] I am willing to submit a pull request to implement this change.
